interpreter/src/main/scala/pl/touk/nussknacker/engine/variables/GlobalVariablesPreparer.scala

```diff
@@ -62,9 +57,12 @@ class GlobalVariablesPreparer(
       case _ =>
         ObjectWithType(
           componentDefWithImpl.implementation,
-          componentDefWithImpl.returnType.getOrElse(
-            throw new IllegalStateException("Global variable with empty return type.")
-          )
+          componentDefWithImpl
+            .asInstanceOf[MethodBasedComponentDefinitionWithImplementation]
```

Comment:
We always had an assumption, that global variables are handled by `MethodBasedComponentDefinitionWithImplementation` - it was just not exposed. This casting is necessary because I removed `returnType` from base `ComponentDefinitionWithImplementation` class - it is only available for `MethodBasedComponentDefinitionWithImplementation` implemenetation. Previous approach was very confusing because for the dynamic components, we can do any transformation of the context of variables - we can clear context, add a new variable, add multiple variables, so why to have one "returnType"? I think it was one of the reasons why most of the developers avoid touching this part of the code :-/

Reply:
ok, i understand, but maybe we could change `GlobalVariablesPreparer.globalVariablesDefWithImpl` to `Map[String, MethodBasedComponentDefinitionWithImplementation]` and in the apply method filter out dynamic components from `expressionDefinition` (or cast/throw exception there, at least it happens earlier and in one place)

Reply:
Yeah, you're right - changed.